### DISCUSSION TITLE
Serial Bridge dropping serial data

### DISCUSSION THREAD

**Discussion by shanekuz** [arendst/Tasmota#13388] [Q&A] [answered]
Im using the serial bridge and am getting communications but it is dropping data.

With my SpaNet serial connection i should issue a RF command and should get a response like this:

RF:
,R2,18,250,51,70,4,13,50,55,19,6,2020,376,9999,1,0,490,207,34,6000,602,23,20,0,0,0,0,44,35,45,:
,R3,32,1,4,4,4,SW V5 17 05 31,SV3,18480001,20000826,1,0,0,0,0,0,NA,7,0,470,Filtering,4,0,7,7,0,0,:
,R4,NORM,0,0,0,1,0,3547,4,20,4500,7413,567,1686,0,8388608,0,0,5,0,98,0,10084,4,80,100,0,0,4,:
,R5,0,1,0,1,0,0,0,0,0,0,1,0,1,0,376,0,3,4,0,0,0,0,0,1,2,6,:
,R6,1,5,30,2,5,8,1,360,1,0,3584,5120,127,128,5632,5632,2304,1792,0,30,0,0,0,0,2,3,0,:
,R7,2304,0,1,1,1,0,1,0,0,0,253,191,253,240,483,125,77,1,0,0,0,23,200,1,0,1,31,32,35,100,5,:
,R9,F1,255,0,0,0,0,0,0,0,0,0,0,:
,RA,F2,0,0,0,0,0,0,255,0,0,0,0,:
,RB,F3,0,0,0,0,0,0,0,0,0,0,0,:
,RC,0,1,1,0,0,0,0,0,0,2,0,0,1,0,:
,RE,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,-4,13,30,8,5,1,0,0,0,0,0,:*
,RG,1,1,1,1,1,1,1-1-014,1-1-01,1-1-01,0-,0-,0,:*

But im getting two lines in console and the same is being sent via MQTT like this:

03:39:38.459 MQT: tele/spanet/RESULT = "SSerialReceived":"RF:\r\n,R2,0,280,48,58,6,15,31,4,17,10,2021,411,9999,1,0,389,126,,R3,10,1,4,4,4,SW V6 19 11 12,SVM2,20340001,20000722,0,0,1,0,0,,,4"}
03:39:38.644 MQT: tele/spanet/RESULT = {"SSerialReceived":"10,1,0,3584,120,127,96,5632,5632,560,2717,0,40,0,0,0,01,6,0,410"}

If i run the RF command again i will get different parts of the full response.

Any Ideas as to why this would happen?  It would be ideal to get each line as a MQTT response or the whole lot as one response.

**Comment by barbudor** [comment, marked as answer]
Hi
Which bit rate are you using?
Software Serial has limited performance above 9600 (and even at 9600). As you seem to have quite a huge amount of data, you may want to consider ESP32 that has 2 hardware serial ports + capability to process serial data in Berry language.

**Reply by shanekuz** [reply]
The unit uses 38400 as a baudrate.  do you think this is the issue?  is that a reason it would drop data?

**Reply by shanekuz** [reply]
I set it up again using just serial tx and rx to the uart pins and could send commands so know it was working but the console got no data back from the unit.  Should this work?  also would that remove the baudrate issues

**Reply by barbudor** [reply]
Definitively 38400 + large packets won't work reliably with Software Serial
You need hardware serial. My preferably for such type of application is ESP32 because you have available hardware UART 1 and 2 while leaving UART0 for the console.
Plus you have Berry language if you want to implement a driver for processing the data on the ESP directly.
